pycbc/inference/sampler/cpnest.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (C) 2016  Collin Capano
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+"""
+This modules provides classes and functions for using the cpnest sampler
+packages for parameter estimation.
+"""
+
+from __future__ import absolute_import
+
+import numpy, os
+import cpnest
+import logging
+import  cpnest.model as cpm
+#from pycbc.pool import choose_pool
+
+from .base import BaseSampler
+from .base_mcmc import (BaseMCMC, raw_samples_to_dict,
+                        get_optional_arg_from_config)
+
+from pycbc.inference.io import (CPNestFile,validate_checkpoint_files)
+from .. import models
+
+
+#
+# =============================================================================
+#
+#                                   Samplers
+#
+# =============================================================================
+#
+
+class CPNestSampler(BaseSampler):
+    """This class is used to construct an NestedSampling sampler from the emcee
+    package's EnsembleSampler.
+
+    Parameters
+    ----------
+    model : model
+        A model from ``pycbc.inference.models``.
+    nlive : int
+        Number of walkers to use in sampler.
+    pool : function with map, Optional
+        A provider of a map function that allows a function call to be run
+        over multiple sets of arguments and possibly maps them to
+        cores/nodes/etc.
+    """
+    name = "cpnest"
+    _io = CPNestFile
+
+    #burn_in_class = MCMCBurnInTests
+
+    def __init__(self, model, nlive, checkpoint_interval=None,
+                 logpost_function=None, nthreads=8, maxmcmc=1000,resume=True):
+
+        self.model = model
+        self.nlive = nlive
+        self.maxmcmc = maxmcmc
+        self.nthreads = nthreads
+        #self.names = model.sampling_params
+        # create a wrapper for calling the model
+        if logpost_function is None:
+            logpost_function = 'logposterior'
```

Comment:
If `logpost_function` isn't being used for anything (which it appears not to be), then remove it.